packages/snaps-execution-environments/src/common/lockdown/lockdown-events.ts

```diff
@@ -0,0 +1,45 @@
+// When creating a sandbox, limitation of the events from leaking
+// sensitive objects is required. This is done by overriding own properties
+// of prototypes of all existing events.
+
+import { logError } from '@metamask/snaps-utils';
+
+/**
+ * Targeted Event objects and properties.
+ * Note: This is a map of the prototypes that inherit from Events with
+ * properties that are identified to leak sensitive objects.
+ */
+export const targetEvents: { [index: string]: any } = Object.freeze({
+  UIEvent: ['view'],
+  MutationEvent: ['relatedNode'],
+  MessageEvent: ['source'],
+  FocusEvent: ['relatedTarget'],
+  MouseEvent: ['relatedTarget', 'fromElement', 'toElement'],
+  TouchEvent: ['targetTouches', 'touches'],
+  Event: ['target', 'currentTarget', 'srcElement', 'composedPath'],
+});
+
+/**
+ * Attenuate Event objects by replacing its own properties.
+ */
+export function executeLockdownEvents() {
+  try {
+    Object.keys(targetEvents).forEach((event) => {
+      const props = targetEvents[event];
+      for (const prop of props) {
+        try {
+          // @ts-expect-error Ignore index signature error.
```

Comment:
try to fix instead ignore

Reply:
The problem with this is that `globalThis` is not an object in our 'scope' of Class definition according to the TypeScript. So, accessing it in many ways is opposite to the TypeScript or ESLint rules. For the TypeScript `globalThis` is considered as a namespace, so it makes it even more difficult to deal with it. Sometimes for security related parts of code with similar issues, we do exactly this with `@ts-expect-error` since there might be no other way around.
I do not have a proper idea on how to fix this. If you have any, let me know, I will try. Thanks.

Reply:
This is now fixed with a solution provided by Maarten.